### PR TITLE
feat(AudioPlayer): Add AudioPlayer component (#993)

### LINKED ISSUE #993: Add `AudioPlayer`
### Description

Make components of an audio player like this one.

https://kolber.github.io/audiojs/

### Problem Statement/Justification

Currently, an audio player must be implemented from scratch.

### Proposed Solution or API

The src attribute should be passed as the path to the audio file or an Audio object.

Allow customization of playback buttons, etc. with attributes.

Update the current time and seek bar in the `timeupdate` event of the audio object.

### Alternatives

_No response_

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by AVAM16**
Closes #993 

## Description

> Add a brief description

Creation of the AudioPlayer component. It can take custom icons, titles, can show the duration and the current time stamp, can be used with a slider to move the timeline, can change background color and border color and can be muted with a mute button.

## Current behavior (updates)

> Please describe the current behavior that you are modifying

## New behavior

> Please describe the behavior or changes this PR adds

### CODE REVIEW

File: packages/components/audio-player/src/audio-player.tsx [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+  const [isPlaying, setIsPlaying] = useState(false)
+  const [durationString, setDurationString] = useState("")
+  const [currTime, setcurrTime] = useState<number>(0)
+  const [duration, setDuration] = useState<number>(0)
+
+  const handleSliderChangeEnd = (value: number) => {
+    file.current.currentTime = value
+  }
+
+  useEffect(() => {
```
Comment (108yen) [inline packages/components/audio-player/src/audio-player.tsx:109]: `useSageLayoutEffect`.

File: packages/components/audio-player/package.json [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    "typecheck": "tsc --noEmit",
+    "prepack": "clean-package",
+    "postpack": "clean-package restore",
+    "gen:docs": "tsx ../../../scripts/generate-docs"
+  },
+  "dependencies": {
+    "@yamada-ui/button": "workspace:^",
+    "@yamada-ui/core": "workspace:*",
+    "@yamada-ui/slider": "workspace:^",
+    "@yamada-ui/utils": "workspace:*"
```
Comment (108yen) [inline packages/components/audio-player/package.json:52]: Do not use yamada-ui components such as `Button` or `Slider`.
(`@yamada-ui/core` and `@yamada-ui/utils` are fine.)

File: packages/components/audio-player/tests/audio-player.test.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    )
+    const playButton = screen.getByRole("button", { name: /play/i })
+    expect(playButton).toBeTruthy()
+    expect(playButton).toContainElement(screen.getByTestId("fa-robot"))
+    fireEvent.click(playButton)
+    const pauseButton = screen.getByRole("button", { name: /pause/i })
+    expect(pauseButton).toBeTruthy()
+    expect(pauseButton).toContainElement(screen.getByTestId("fa-robot"))
+    fireEvent.click(pauseButton)
+    await a11y(container)
```
Comment (108yen) [inline packages/components/audio-player/tests/audio-player.test.tsx:122]: The `a11y` test should only be done once.

**Review by 108yen — CHANGES_REQUESTED** [review]
Will you add other features such as `onPlay` as linked in issue #993?

Finally, yamada check is also made.

**Comment by AVAM16** [comment]
Given that I cannot use Yamada-UI components such as Button and Slider, I do not think it is worth to continue this PR. The entire feature revolved around using components from the same project. Thank you.

**Pull request closed without merging**